client/interfaces.go

```diff
@@ -0,0 +1,150 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+// Plug represents a capacity offered by a snap.
+type Plug struct {
+	Name string `json:"name"`
+	Snap string `json:"snap"`
+	// NOTE: json format intentionally using old "skill" terminology
+	Interface string                 `json:"type,omitempty"`
+	Attrs     map[string]interface{} `json:"attrs,omitempty"`
+	Apps      []string               `json:"apps,omitempty"`
+	Label     string                 `json:"label,omitempty"`
+}
+
+// Slot represents the potential of a given snap to connect to a given plug.
+type Slot struct {
+	Name string `json:"name"`
+	Snap string `json:"snap"`
+	// NOTE: json format intentionally using old "skill" terminology
+	Interface string                 `json:"type,omitempty"`
+	Attrs     map[string]interface{} `json:"attrs,omitempty"`
+	Apps      []string               `json:"apps,omitempty"`
+	Label     string                 `json:"label,omitempty"`
+}
+
+// PlugConnections represents a single plug and slots that are connected to it.
+type PlugConnections struct {
+	Plug
+	// NOTE: json format intentionally using old "skill" terminology
+	Connections []Slot `json:"granted_to"`
```

Comment:
This is not actually a Slot, though, as it misses pretty much all details. It should be something like SlotRef.

Fine to do that in the future, rather than during rename.